test/e2e/index.test.ts

```diff
@@ -194,5 +194,20 @@ describe('e2e', () => {
         ),
       ).to.contain('Mui-selected');
     });
+
+    it('should not scroll when changing the selected row', async () => {
+      await renderFixture('DataGrid/RowSelection');
+      await page.click('[role="cell"][data-rowindex="0"]');
+      await page.evaluate(() =>
+        document.querySelector('[role="cell"][data-rowindex="3"]')!.scrollIntoView(),
+      );
+      expect(
+        await page.evaluate(() => document.querySelector('.MuiDataGrid-window')!.scrollTop!),
+      ).not.to.equal(0);
+      await page.click('[role="cell"][data-rowindex="3"]');
+      expect(
+        await page.evaluate(() => document.querySelector('.MuiDataGrid-window')!.scrollTop!),
+      ).not.to.equal(0);
```

Comment:
Should the `document.querySelector('.MuiDataGrid-window')!.scrollTop!` be equal between the two `expect` ?
If so, we may want to check more strictly if the two are equal, and not just if they are not equal to 0